src/main/java/bgu/spl/mics/application/passiveObjects/Squad.java

```diff
@@ -1,87 +1,129 @@
+     * Initializes the squad. This method adds all the agents to the squad.
+     * <p>
+     *
+     * @param agents Data structure containing all data necessary for initialization of the squad.
+     *
+     * @pre none
+     * @post (for agent : agents { getAgents ( agent.getSerialNumber) == true})
+     */
+    public void load(Agent[] agents) {
+        for (Agent agent : agents) {
+            this._agents.put(agent.getSerialNumber(), agent);
+        }
+    }
 
-	/**
-	 * simulates executing a mission by calling sleep.
-	 * @param time   milliseconds to sleep
-	 * @pre (for serial : serials {getAgents([serial]).state == BLOCKED})
-	 * @post (current_time >= start_time + time && getAgents(serials) == true)
-	 */
-	public void sendAgents(List<String> serials, int time){
-		// TODO Implement this
-	}
+    /**
+     * Releases agents.
+     *
+     * @pre (for serial : serials { getAgents ( [serial]).state == BLOCKED})
+     * @post (getAgents ( serials) == true)
+     */
+    public void releaseAgents(List<String> serials) {
+        Agent agent;
+    	for (String serial : serials) {
+            agent = _agents.get(serial);
+            agent.release();
+            agent.notifyAll();
+        }
+    }
+
+    /**
+     * simulates executing a mission by calling sleep.
+     *
+     * @param time milliseconds to sleep
+     *
+     * @pre (for serial : serials { getAgents ( [serial]).state == BLOCKED})
+     * @post (current_time > = start_time + time & & getAgents ( serials) == true)
+     */
+    public void sendAgents(List<String> serials, int time) {
+        try {Thread.sleep(time);} catch (InterruptedException e) {}
+        releaseAgents(serials);
+    }
 
-	/**
-	 * acquires an agent, i.e. holds the agent until the caller is done with it
-	 * @param serials   the serial numbers of the agents
-	 * @return ‘false’ if an agent of serialNumber ‘serial’ is missing, and ‘true’ otherwise
-	 * @pre none
-	 * @post (for serial : serials {getAgents([serial]).state == BLOCKED})
-	 */
-	public boolean getAgents(List<String> serials){
-		// TODO Implement this
-		return false;
-	}
+    /**
+     * acquires an agent, i.e. holds the agent until the caller is done with it
+     *
+     * @param serials the serial numbers of the agents
+     *
+     * @return ‘false’ if an agent of serialNumber ‘serial’ is missing, and ‘true’ otherwise
+     *
+     * @pre none
+     * @post (for serial : serials { getAgents ( [serial]).state == BLOCKED})
+     */
+    public boolean getAgents(List<String> serials) {
+		Agent agent;
+		// First, make sure all agent exists in the squad, to avoid acquiring and releasing.
+		for (String serial : serials) {
+			agent = _agents.get(serial);
+			if (agent == null)
+				return false;
+		}
+		// after validating all agent exists in the
+		for (String serial : serials) {
+			agent = _agents.get(serial);
+			synchronized (agent) {
+				while (!agent.isAvailable()) {
+					try {wait();} catch (InterruptedException e) {}
```

Comment:
I think this should be `agent.wait()` to match the `agent.notifyAll()` in `releaseAgents()`